ignite/distributed/utils.py

```diff
@@ -427,21 +432,21 @@ def run(local_rank, *args, **kwargs):
     ComputationModel._ext_local_rank = index
 
 
-def _set_model(model, temporary=False):
+def _set_model(model: Any, temporary: bool = False) -> None:
     global _model, _need_to_sync
     _model = model
     _need_to_sync = True
     if not isinstance(_model, _SerialModel) and not temporary:
         _need_to_sync = False
 
 
-def _assert_backend(backend):
+def _assert_backend(backend: Optional[str]) -> None:
```

Comment:
Same here for `Optional` ?

Reply:
same as above, because in the `__init__` function of `Parallel` `backend` is `None` by default
https://github.com/pytorch/ignite/blob/51067381b457c098b8dd80972ecfb8ab113baef2/ignite/distributed/launcher.py#L176-L185
and in the function `run` it calls `utils.spwan`
https://github.com/pytorch/ignite/blob/51067381b457c098b8dd80972ecfb8ab113baef2/ignite/distributed/launcher.py#L271
and in the whole `Parallel` `self.backend` is not set with a new backend, therefore it could be `None` till this point.

Reply:
I think there is some to fix in the code as here it shouldn't be optional and can't be None.
And `ignite/ignite/distributed/launcher.py` does not use `_assert_backend`. It is used only in
https://github.com/pytorch/ignite/blob/b977b746edcbec9c29b8b88f40157f7d7db2601f/ignite/distributed/utils.py#L449
where backend is not optional.

Reply:
No, it is also used in the `spawn` function
https://github.com/pytorch/ignite/blob/b977b746edcbec9c29b8b88f40157f7d7db2601f/ignite/distributed/utils.py#L192-L199
https://github.com/pytorch/ignite/blob/b977b746edcbec9c29b8b88f40157f7d7db2601f/ignite/distributed/utils.py#L306
there is a very long function description, which I didn't want to add here 😄

Reply:
Yes, true, that's why it was refactored into `_assert_backend`, but anyway in both `backend` can't be optional.

Reply:
ok, is it then ok to add here `... and self.backend is not None:`?
https://github.com/pytorch/ignite/blob/b977b746edcbec9c29b8b88f40157f7d7db2601f/ignite/distributed/launcher.py#L276-L278

Reply:
otherwise I can also put here `# type: ignore`